pages/search/index.tsx

```diff
@@ -235,12 +234,10 @@ export default function Search({
  * relevant search results on the server side (via fetchResults).
  *
  */
-export async function getServerSideProps({ resolvedUrl, req }) {
+export async function getServerSideProps({ resolvedUrl, req, query }) {
   const bannerNotification = process.env.SEARCH_RESULTS_NOTIFICATION || ""
 
-  // Remove everything before the query string delineator '?', necessary for correctly parsing the 'q' param.
-  const queryString = resolvedUrl.slice(resolvedUrl.indexOf("?") + 1)
-  const results = await fetchResults(mapQueryToSearchParams(parse(queryString)))
```

Comment:
The `qs` package was introduced to help parse the request URL query which is helpful but the `collapseMultiValueQueryParams` function (which is called by `mapQueryToSearchParams`)  does not expect a fully parsed object.

The first object below is the type of keys it expects when it parses the queryParams. The second object is what `qs`'s "parse' function returns.

I decided to _not_ use the "parse" function because nextjs returns the same object in `getServerSideProps` and through the `useRouter` function. Additionally, "parse" cannot be used for the object that `useRouter` returns.

![Screenshot 2024-05-03 at 11 11 24 AM](https://github.com/NYPL/research-catalog/assets/1280564/4786053b-46ca-4b13-a284-ec300eee1a8c)